File: day15/dueling_generators.py
#!/usr/bin/env python3

#gen = [65, 8921]
gen = [116, 299]
FACT = [16807, 48271]
# PAIRS = 5
PAIRS = 40000000
cnt = 0

for _ in range(PAIRS):
    for i in range(2):
        gen[i] = (gen[i] * FACT[i]) % 2147483647
        
    if (gen[0] & 0xffff) == (gen[1] & 0xffff):
        cnt += 1

print("result1:", cnt)

File: day15/dueling_generators.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);

# my @gen = (65, 8921); # test
my @gen = (116, 299);
my @fact = (16807, 48271);
my $PAIRS = 40_000_000;
my $cnt = 0;

for (1..$PAIRS) {
    $gen[$_] = ($gen[$_] * $fact[$_]) % 2147483647 for 0..1;
    $cnt++ if ($gen[0] & 0xffff) == ($gen[1] & 0xffff);
}

say "result1: $cnt";
